ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/CommitContext.scala

```diff
@@ -37,7 +41,10 @@ private[kvutils] trait CommitContext {
   /** Retrieve value from output state, or if not found, from input state. */
   def get(key: DamlStateKey): Option[DamlStateValue] =
     outputs.get(key).orElse {
-      inputs.getOrElse(key, throw Err.MissingInputState(key))
+      val value = inputs.getOrElse(key, throw Err.MissingInputState(key))
+      if (accessedInputKeys.add(key))
```

Comment:
We don't need to keep track of the read order just the set of read keys. I.e., we could make the resulting `Iterable` deterministic by simply using a `SortedSet`.

Reply:
Insertion order should be deterministic too (because access order is deterministic due to pre-execution and post-execution being deterministic on a given submission) and is slightly more efficient with a `LinkedHashSet`. @andreaslochbihler-da could you validate my reasoning?

Reply:
Also, not sure if we really need deterministic or stable iteration over the input keys at all; do we already know that?

Reply:
We don't even need deterministic order for the read set, we only care about membership and nothing else in the end.

Reply:
So no reliance by any ledger on any of these properties? If so yes and I could also just use an `HashSet` and save some more memory.

Reply:
Not that I know of. It's easy to fix this in the actual execution engine implementation if needed.

Reply:
Done, can you have a further look?

Reply:
Looks good to me.

Reply:
Thanks!

Reply:
For pre-execution, we must compute the same serialization for the read set on all replicas. This is because the read set contributes to the hash the replicas need to each consensus on. So we will ultimately need a deterministic order on the keys in the read set. For now, using any `Set` is fine as long as we recover determinism before/during serialization, e.g., by ordering all keys in it.

Reply:
OK, then we might want to revisit this because always iterating in access order is cheap here and spares us reordering later. Added a note to the LLD and will also cross-link this discussion.